src/pages/images/ImageSelectorModal.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { FC } from "react";
+import { Modal } from "@canonical/react-components";
+import { RemoteImage } from "types/image";
+import RemoteImageSelector from "pages/images/RemoteImageSelector";
+import UploadIsoImage from "pages/storage/UploadIsoImage";
+
+interface Props {
+  onClose: () => void;
+  onSelect: (image: RemoteImage, type: string | null) => void;
+}
+
+const REMOTE_IMAGES = "remoteImages";
+const UPLOAD = "upload";
+
+const ImageSelector: FC<Props> = ({ onClose, onSelect }) => {
+  const [content, setContent] = React.useState(REMOTE_IMAGES);
+  return (
+    <Modal
+      close={onClose}
+      title={
+        content === REMOTE_IMAGES ? "Select base image" : "Upload custom ISO"
+      }
+      className={content === REMOTE_IMAGES ? "image-select-modal" : null}
+    >
+      {content === REMOTE_IMAGES && (
+        <RemoteImageSelector
+          onSelect={onSelect}
+          onUpload={() => setContent(UPLOAD)}
+        />
+      )}
+      {content === UPLOAD && (
+        <UploadIsoImage
+          onCancel={() => setContent(REMOTE_IMAGES)}
```

Comment:
Is there a way to actually cancel the current upload operation? So that if a user happens to change their mind while uploading, they can stop the upload for good instead of having it continue in the background?